libs/datamodel/core/tests/validation/attributes/schema/bad_schema_attribute.prisma

```diff
@@ -0,0 +1,47 @@
+datasource testds {
+    provider = "postgresql"
+    url = env("TEST_DATABASE_URL")
+    schemas = ["public"]
+}
+
+generator js {
+    provider = "prisma-client-js"
+    previewFeatures = ["multiSchema"]
+}
+
+model Test {
+  id Int @id
+  @@schema
+}
+
+enum Language {
+  English
+  Spanish
+
+  @@schema
+}
+
+model Test2 {
+  id Int @id
+  @@schema(101)
+}
+
+
+// [1;91merror[0m: [1mArgument "map" is missing.[0m
+//   [1;94m-->[0m  [4mschema.prisma:14[0m
+// [1;94m   | [0m
+// [1;94m13 | [0m  id Int @id
+// [1;94m14 | [0m  [1;91m@@schema[0m
+// [1;94m   | [0m
```

Comment:
Isn't it mapping a `name`? I thought `map` was optional for names that are not valid identifiers in PSL.

Reply:
This one is tough, I gave it some thought and landed on `map: ` because `name: ` usually means the name of the item in the context of the Client, and `map` means the name in the database. We can easily change this to `name:` if you feel it's better.

Reply:
I think we have both actually:
`@@schema(name: "foobar", map: "9foobar")`

The missing value is the `"foobar"` here which can also be used without the `name:` bit.

Reply:
There is no schema on the client, so the way I implemented it, only `@@schema("foobar")` or `@@schema(map: "foobar")` is valid. We could allow `@@schema(name: "foobar")` (instead of `map:`) but I don't think we could make sense of both map and name at the moment — wouldn't name be completely unused, since it doesn't correspond to anything in the client api?

Reply:
This is indeed an interesting case which we never had before I think.

`"foobar"` has to have a key connected to it - we have that in all other cases.
`map: "9-foobar"` is a completely different thing in my opinion that is not connected to the case you are trying to test here.

I will see if I can find any ideas in our docs...

Reply:
We only have a hand full of model attributes for now:

`@@id` - https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#id-1
`@@unique` - https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#unique-1
`@@index` - https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#index
`@@map` - https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#map-1

All these have a argument which can be either named or without:
- 3 call it `fields`
- For `@@map` we have `name` as the argument that can be named or not.

Now via this feature we are introducing `@@schema` which also has a required argument that defines what the thing is called in the database by default, where we seem to be lacking a term to call it for now.

Usually the "databasename" is by default from e.g. the field name, model name or similar and we need to be able to overwrite that with `(@)(@)map` (`@@map` for a model, `@map` for fields and enums, and `map` for e.g. `@@index` and similar)

Or am I missing something? Would we _only_ have `map` for those invalid identifiers?

Reply:
all attributes in PSL have a default argument whose name can be omitted. For example @@index(fields: [a, b, c], name: "abcdidx") is the same as @@index([a, b, c], name: "abcidx"). The default argument is called `name` in many cases (relations come to mind), but it's not universal.

Reply:
Oops, I posted before I saw your comment (just refreshed and it popped up)

Reply:
Same for me 😆

Reply:
The way I think of `map` is that it lets you define the name of the name of _something_ in the database (index, field, model...). It is mostly useful when it wouldn't be a valid PSL identifier (otherwise it defaults to the PSL name), but not only. Since these schema names are those defined in the database, that's what I went with, and `name: "abcd"` would be the name in the client API, in case the database name doesn't work for it. It can feel kind of backwards, since there is no PSL identifier (without quote) for schemas. For contrast, my postgres extensions proposal makes room for both a PSL identifier and a mapped name for each extension (`extensions =  [uuidOssp(map: "uuid-ossp")]`).

Reply:
I think we don't need PSL identifiers for schemas, since it would be a level of indirection that we don't need for what we envision at the moment, but it might be something we want in the future, and then we'd have to find a migration path.

Reply:
Yeah I think that is what I just realized as well. The only thing we have is the `map` as we have a string both in `@@schema` and `schemas` array.

Still weird how this introduces another case somehow.
On the other hand `@@index(map: ...)` is not _really_ different besides that we have a default which causes the `map` not to be rendered.

Reply:
So we should think about this carefully for the upcoming PSL proposal. For the immediate future, do you think we can move forward with this? I can also very easily change the default argument from `map` to a `name`, whatever sounds better.

Reply:
I _think_ that `map` is correct here, but it indeed raises the question how this fits into what we did until now and also highlights how "different" this `@@schema` really is from the things we had before. It is similar to `@@map` in that it qualifies a model/enum further, but also different.

Reply:
Yes, I think it's the first case where we have an attribute that expresses "belongs to" (it could be belonging to a schema, to a datasource, to a role, to a partition, ...)

Reply:
I'm going to merge in order to unblock the client team. Let's keep this problem in mind and address it in the upcoming PSL proposal.